### PR TITLE
Fix package path computation in ClasspathScanner

### LINKED ISSUE #2500: ClasspathScanner.getRootUrisForPackage returns empty list when package name conflicts with module name
When developing a multi-module project that has similar package names it is possible that `getRootUrisForPackage` will fail to return a valid list. 

## Steps to reproduce

Project1.

```java
module io.litterat.pep {
	exports io.litterat.pep;
	exports io.litterat.pep.mapper;
}
```

Project2.

```java
open module io.litterat.pep.test {
        exports io.litterat.pep.test;
	requires io.litterat.pep;
	requires org.junit.jupiter.api;
}
```

In eclipse (2020-09,2020-12 and other versions) when selecting the package `io.litterat.pep.test` no JUnit 5 tests are found or executed. In 2020-12 no errors or console output is shown. 

The root cause is that `getRootUrisForPackage( "io.litterat.pep.test" )` results in calling `getClassLoader().getResources( "io/litterat/pep/test" )`. This results in calling `ClassLoaders$BootClassLoader findResources` which calls `Resources.toPackageName("io/litterat/pep/test" )` which results in package name `"io.litterat.pep"` (the name of the first module). The loader then resolves the name to the wrong `LoadedModule` which does not have the selected package. This results in no resources being returned.

Using Eclipse and attempting to execute all tests on either source directory or project results in all tests in the `io.litterat.pep.test` package being skipped. However, tests in sub-packages (e.g. `io.litterat.pep.test.extra`) are executed correctly.

Changing the name of the package so that the second project does not use the other module package as a base corrects the problem (e.g. `io.litterat.peptest`).

## Context

 - Used versions (Jupiter/Vintage/Platform): junit-platform-commons-1.7.0.jar
 - Build Tool/IDE: Eclipse

## Deliverables

It's not clear if this is a JDK bug or an issue with the way `getResources` is being called. Given that no output or feedback is provided this has taken quite a long time to figure out. Either issue should be documented or fixed.

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Prior to this commit no trailing `/` character was appended to the computed package path.

Now, except for the default package `""`, a `/` is appended to package path. This leads to corrected and documented behavior even if two modules start with the same name elements.

Fixes #2500
---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java [hunk 1]
```diff
@@ -181,21 +181,28 @@ void scanForClassesInPackage() {
...
-		assertThat(classNames).hasSize(1).contains("com.greetings.Main");
+		var classpathScanner = new ClasspathScanner(() -> layer.findLoader(root), ReflectionUtils::tryToLoadClass);
+		{
+			var classes = classpathScanner.scanForClassesInPackage("com.greetings", allClasses);
+			var classNames = classes.stream().map(Class::getName).collect(Collectors.toList());
+			assertThat(classNames).hasSize(2).contains("com.greetings.Main", "com.greetings.test.Tests");
+		}
+		{
+			var classes = classpathScanner.scanForClassesInPackage("com.greetings.test", allClasses);
+			var classNames = classes.stream().map(Class::getName).collect(Collectors.toList());
```
Comment (marcphilipp) [inline platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java:203]: nit: I'd use a static import for `toList()`.

**Review by marcphilipp — APPROVED** [review]

**Review by sbrannen — COMMENTED** [review]
How was the `platform-tests/src/test/resources/com.greetings.test@0-ea.jar` file created; what does it contain; and can we reproduce/modify it?

File: platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java (hunk 1 above)
Comment (sormuras) [reply platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java:203]: Let's wait until we switched to Java 16 -- and replace all those with `Stream::toList`

https://twitter.com/sormuras/status/1337495352962904066

**Comment by sormuras** [comment]
> How was the `platform-tests/src/test/resources/com.greetings.test@0-ea.jar` file created; what does it contain; and can we reproduce/modify it?

Good questions. I'll replace this JAR file and the related `com.greetings@1-ea.jar` one with a on-the-fly compilation solution.

Btw. do we know how to reproduce/modify `gh-1436-invalid-nested-class-file.jar` and `jartest.jar`?

File: platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java (hunk 1 above)
Comment (sormuras) [reply platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java:203]: All? All 44 usages found in test files.

image

Comment (marcphilipp) [reply platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java:220]: Go for it! 👍

**Review by marcphilipp — APPROVED** [review]
👍

**Comment by sbrannen** [comment]
> Btw. do we know how to reproduce/modify `gh-1436-invalid-nested-class-file.jar` and `jartest.jar`?

I don't know. Sounds like it may be problematic. 

🤷‍♂️

**Comment by sbrannen** [comment]
>  I'll replace this JAR file and the related `com.greetings@1-ea.jar` one with a on-the-fly compilation solution.

Thanks! 👍

**Review by sbrannen — APPROVED** [review]

**Pull request merged**